wandb/sdk/wandb_run.py

```diff
@@ -2632,6 +2632,10 @@ def link_artifact(
                     entity,
                     project,
                 )
+                if artifact._ttl_changed:
```

Comment:
We want to show the warning if the artifact has a TTL rule, i.e.:
```suggestion
                if not artifact._ttl_is_inherited:
```